connpool.go

```diff
@@ -0,0 +1,64 @@
+// Package goconnpool implements connections pool with ratelimits and backoff for broken connections.
+//
+// Connection returned by the pool is protocol-independent.
+//
+package goconnpool
+
+import (
+	"context"
+	"fmt"
+	"net"
+)
+
+var (
+	errRatelimited  = fmt.Errorf("ratelimited")
+	errServerIsDown = fmt.Errorf("server is down")
+)
+
+// Conn is a wrapper around net.Conn interface
+type Conn interface {
+	net.Conn
+
+	// MarkBroken marks connection not usable any more:
+	// connection will be closed when Close() method will be invoked.
+	// Otherwise connection will be returned into connections pool when Close function will be called.
+	MarkBroken()
+}
+
+// ConnPool is the base interface to interact with user.
+type ConnPool interface {
+	// OpenConnNonBlock requests one connection from the pool.
+	//
+	// If the pool already contains opened connection, this connection will be returned.
+	//
+	// If each registered server is down, function returns an error.
+	// Otherwise function returns active connection (to some alive server in round-robin order)
+	// which could be used to send any type of request.
+	//
+	// Connection should be closed after usage. This action will return connection into pool.
```

Comment:
Strange comment or I dont understand smth.
Why connection is going to be 'closed'? Does it mean that you open connection, use it and close after all? Or you should just return connection to the pool without really closing this connection?

Reply:
Fixed. Close at now just closes the connection, ReturnToPool returns the connection to pool %)